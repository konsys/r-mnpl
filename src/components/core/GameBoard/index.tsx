import React from "react";

import { useStore } from "effector-react";

import { IFieldModalPosition } from "../../../types/types";
import { GameBoardView } from "./GameBoardView";

export const fieldsActionPosition = (): IFieldModalPosition => {
  return {
    left: 0,
    top: 0,
  };
};

export const GameBoard = () => {
  return <GameBoardView />;
};
