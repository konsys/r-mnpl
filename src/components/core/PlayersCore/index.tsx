import React from "react";
import { RouteComponentProps } from "@reach/router";

interface Props extends RouteComponentProps {}

//TODO unused now, is it needed?
export const Game = (props: Props) => {
  return <>GAME</>;
};
