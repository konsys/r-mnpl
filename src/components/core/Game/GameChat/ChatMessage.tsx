import { Chip, Grid, Typography } from "@material-ui/core";

import { IUser } from "../../../../types/types";
import React from "react";
import Reply from "@material-ui/icons/Reply";
import StarOutlined from "@material-ui/icons/StarOutlined";
import { addReplyToEvent } from "../../../../stores/Game/GameChatStore";
import moment from "moment";
import { theme } from "../../../../theme";
import { useStore } from "effector-react";
import { userStore } from "../../../../stores/Game/UserStore";

export interface IChatMessage {
  fromUser: IUser;
  replies?: IUser[];
  message: string;
  time: Date;
}

export interface IChatMessageProps extends IChatMessage {}

export default function ChatMessage(props: IChatMessageProps) {
  const user = useStore(userStore);
  return (
    <>
      <Grid
        container
        alignContent="space-between"
        alignItems="center"
        justify="center"
        spacing={1}
      >
        <Grid item>{moment(props.time).format("HH:mm").toString()}</Grid>{" "}
        <Grid item>
          <Typography variant="body2">
            <Reply
              onClick={() => {
                addReplyToEvent(props.fromUser);
              }}
              style={{ width: "15px", height: "15px", cursor: "pointer" }}
            />
          </Typography>
        </Grid>
        <Grid item>
          <Chip
            color={props.fromUser.vip ? "secondary" : "default"}
            size="small"
            label={props.fromUser.name}
            icon={
              props.fromUser.vip ? (
                <StarOutlined
                  style={{
                    width: "15px",
                    height: "15px",
                    color: "white",
                  }}
                />
              ) : undefined
            }
            style={{
              color: props.fromUser.vip ? "white" : theme.palette.text.primary,
            }}
          />
        </Grid>
        {props.replies && props.replies.length ? (
          <Grid item>
            <Typography variant="body2">-</Typography>
          </Grid>
        ) : (
          ""
        )}
        {props.replies && (
          <Grid item>
            {props.replies.map((v, k) => (
              <>
                <Chip
                  key={k}
                  color={
                    v.vip
                      ? "secondary"
                      : v.userId === user.userId
                      ? "primary"
                      : "default"
                  }
                  size="small"
                  label={v.name}
                  icon={
                    v.vip ? (
                      <StarOutlined
                        style={{
                          width: "15px",
                          height: "15px",
                          color: "white",
                        }}
                      />
                    ) : undefined
                  }
                  style={{
                    color: v.vip ? "white" : theme.palette.text.primary,
                  }}
                />
                ,
              </>
            ))}
          </Grid>
        )}
        <Grid item>-</Grid>
        <Grid item>{props.message}</Grid>
        <Grid item></Grid>
      </Grid>
    </>
  );
}
