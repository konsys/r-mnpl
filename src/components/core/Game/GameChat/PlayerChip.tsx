import { Chip } from "@material-ui/core";
import FaceIcon from "@material-ui/icons/Face";
import React from "react";
export default function PlayerChip({
  handleDelete,
  name,
}: {
  handleDelete: () => any;
  name: string;
}) {
  return (
    <>
      <Chip
        size="small"
        icon={<FaceIcon />}
        label={name}
        onDelete={handleDelete}
        variant="outlined"
        color="default"
      />
    </>
  );
}
