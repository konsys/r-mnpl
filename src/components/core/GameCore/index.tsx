import React from "react";
import { RouteComponentProps } from "@reach/router";
import "./style.scss";
import { Chat } from "../../views/Chat/Chat";
import { Contract } from "../../views/Contract/Contract";
import { TableHelper } from "../../views/TableHelper/TableHelper";
import { Tokens } from "../../views/Tokens/Tokens";
import { Ticket } from "../../views/Ticket/ticket";
import { Arbitr } from "../../views/Arbitr/Arbitr";
import { M1tv } from "../../views/M1tv/M1tv";
import { BoardModal } from "../../views/BoardModal/BoardModal";
import { Dices } from "../../views/Dices/Dices";
import { GameLoading } from "../../views/GameLoading/GameLoading";
import { useStore } from "effector-react";
import { BoardCore } from "../BoardCore/BoardCore";
import { UsersCore } from "../PlayersCore/PlayersCore";
import { actionsStore } from "../../../stores/ActionStore";
import { IncomeMessageType } from "../../../types/BoardTypes";
import "react-toastify/dist/ReactToastify.css";
import { ToastContainer } from "react-toastify";

interface Props extends RouteComponentProps {}

export const Game = (props: Props) => {
  const actionState = useStore(actionsStore);
  const actionType = actionState.event.action.type;

  const isModalAction =
    actionType === IncomeMessageType.INCOME_AUCTION_MODAL ||
    actionType === IncomeMessageType.INCOME_CAN_BUY_MODAL ||
    actionType === IncomeMessageType.INCOME_ROLL_DICES_MODAL ||
    actionType === IncomeMessageType.INCOME_TAX_PAYING_MODAL ||
    actionType === IncomeMessageType.INCOME_UN_JAIL_MODAL;

  return (
    <>
      <div className="wrapper" style={{ width: "100%", height: "100%" }}>
        <div className="table _shakehack">
          <div className="table-body">
            <UsersCore />
            <div className="table-body-board">
              <BoardCore />
              <div className="table-body-board-center">
                <M1tv />
                {actionState && (
                  // TODO check for user credentials
                  // userState.userId === actionState.event.action.userId &&
                  <BoardModal isModal={isModalAction} />
                )}
                <Arbitr />
                <Ticket />
                <Chat />
              </div>
              <div className="table-body-board-tokens">
                <Tokens />
              </div>
              <Dices />
              <Contract />
              <TableHelper />
            </div>
          </div>

          <div id="placeholder_gameover"></div>
        </div>
        <div className="table-jokes"></div>
      </div>
      <GameLoading />
      <ToastContainer />
    </>
  );
};
