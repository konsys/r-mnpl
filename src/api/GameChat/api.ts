import { IChatMessage } from "components/core/Game/GameChat/ChatMessage";
import { IChatMessageRequest } from "stores/Game/Chat/GameChatModel";
import { client } from "http/client";

const url = `/chat`;
export const fetchChat = async ({
  message,
  replies,
}: IChatMessageRequest): Promise<IChatMessage[]> => {
  return await (await client.post(url, { message, replies })).data;
};
