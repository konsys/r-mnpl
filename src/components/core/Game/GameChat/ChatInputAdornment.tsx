import { deleteReplyToEvent, repliesTo$ } from "stores/Game/Chat/GameChatModel";

import { InputAdornment } from "@material-ui/core";
import PlayerChip from "./PlayerChip";
import React from "react";
import { useStore } from "effector-react";

export default function ChatInputAdornment() {
  const replyTo = useStore(repliesTo$);
  return (
    <div>
      <InputAdornment position="start" variant="outlined">
        {replyTo && replyTo.users
          ? replyTo.users.map((v, k) => (
              // TODO make standart spacing
              <div style={{ marginRight: "5px" }} key={k}>
                <PlayerChip
                  key={k}
                  handleDelete={() => deleteReplyToEvent(v)}
                  name={v.name}
                />
              </div>
            ))
          : ""}
      </InputAdornment>
    </div>
  );
}
