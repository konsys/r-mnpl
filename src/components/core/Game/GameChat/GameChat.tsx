import {
  Grid,
  Input,
  InputAdornment,
  Switch,
  Typography,
} from "@material-ui/core";
import React, { useEffect, useRef, useState } from "react";
import {
  chatStore,
  deleteReplyToEvent,
  replyStore,
  resetChatEvent,
  resetReplyToEvent,
  sendChatMessageEffect,
} from "../../../../stores/Game/models/GameChatStore";

import ChatMessage from "./ChatMessage";
import { GRID_SPACING } from "../../../../theme";
import PlayerChip from "./PlayerChip";
import { getUserEffect } from "../../../../stores/Game/UserStore";
import { useStore } from "effector-react";
import { useTranslation } from "react-i18next";

enum KeyName {
  "ENTER" = 13,
}

export default function GameChat() {
  const { t } = useTranslation();
  const messages = useStore(chatStore);
  const replies = useStore(replyStore);
  const [m, setM] = useState<string>("");

  useEffect(() => {
    getUserEffect("me");
    return () => {
      resetChatEvent();
    };
  }, []);

  sendChatMessageEffect.done.watch(() => {
    resetReplyToEvent();
    setM("");
  });
  const inputEl = useRef<HTMLInputElement>(null);

  return (
    <>
      <Grid container direction="column" spacing={GRID_SPACING}>
        <Grid item id="game-chat-header">
          <Grid container spacing={GRID_SPACING} alignItems="center">
            <Grid item>
              <Typography variant="h6">{t("Chat")}</Typography>
            </Grid>
            <Grid item>
              <Typography variant="body2"> {123456}</Typography>{" "}
              <Typography variant="body2">{t("now online")}</Typography>
            </Grid>
            <Grid item>
              <Switch
                checked={true}
                onChange={() => null}
                name="checkedA"
                inputProps={{ "aria-label": "secondary checkbox" }}
                color={"primary"}
              />
              <Typography variant="body2">{t("Show bots messages")}</Typography>
            </Grid>
          </Grid>
        </Grid>
        <Grid
          item
          id="game-chat-content"
          style={{
            maxHeight: "225px",
            overflow: "auto",
            scrollBehavior: "smooth",
          }}
        >
          <Grid
            container
            spacing={1}
            direction={"column"}
            justify={"center"}
            alignItems={"flex-start"}
          >
            {messages && messages.length
              ? messages.map((v, k) => (
                  <Grid item key={k}>
                    <ChatMessage
                      key={k}
                      fromUser={v.fromUser}
                      replies={v.replies}
                      message={v.message}
                      time={v.time}
                    />
                  </Grid>
                ))
              : ""}
          </Grid>
        </Grid>
        <Grid item id="game-chat-input">
          <Input
            style={{ width: "100%" }}
            placeholder={t("Type message and press Enter")}
            startAdornment={
              <InputAdornment position="start" variant="outlined">
                {replies && replies.users
                  ? replies.users.map((v, k) => (
                      // TODO make standart spacing
                      <div style={{ marginRight: "5px" }}>
                        <PlayerChip
                          handleDelete={() => deleteReplyToEvent(v)}
                          name={v.name}
                        />{" "}
                      </div>
                    ))
                  : ""}
              </InputAdornment>
            }
            value={m}
            onChange={(e: any) => {
              setM(e.target.value);

              try {
                if (inputEl && inputEl.current && inputEl?.current.focus) {
                  inputEl.current.focus();
                }
              } catch (er) {}
            }}
            onKeyPress={(e: any) => {
              const message = e.target.value;
              if (
                (e.keyCode === KeyName.ENTER || e.which === KeyName.ENTER) &&
                message.length
              ) {
                sendChatMessageEffect({
                  message: message,
                  replies: replies ? replies.users : [],
                });
              }
            }}
          />
        </Grid>
      </Grid>
    </>
  );
}
