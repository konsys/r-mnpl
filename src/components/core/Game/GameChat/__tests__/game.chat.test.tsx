import GameChat from "../GameChat";
import React from "react";
import { createMount } from "@material-ui/core/test-utils";
import { mount } from "enzyme";

describe("Game chat test", () => {
  let shall: typeof mount;

  beforeEach(() => jest.clearAllMocks());
  beforeAll(() => {
    shall = createMount();
  });

  afterAll(() => {
    // @ts-ignore
    shall.cleanUp();
  });

  it("should render", () => {
    expect(shall(<GameChat />)).toMatchSnapshot();
  });
});
