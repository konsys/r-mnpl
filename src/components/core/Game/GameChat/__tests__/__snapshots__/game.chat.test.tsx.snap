// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game chat test should render 1`] = `ReactWrapper {}`;
